src/java/org/apache/cassandra/io/sstable/format/trieindex/TrieIndexSSTableReader.java

```diff
@@ -256,7 +261,7 @@ protected RowIndexEntry getPosition(PartitionPosition key, Operator op, boolean
 
             try (PartitionIndex.Reader reader = partitionIndex.openReader())
             {
-                return reader.ceiling(searchKey, (pos, assumeNoMatch, compareKey) -> retrieveEntryIfAcceptable(searchOp, compareKey, pos, assumeNoMatch));
+                return reader.ceiling(searchKey, (pos, assumeNoMatch, compareKey) -> retrieveEntryIfAcceptable(searchOp, compareKey, pos, assumeNoMatch, forceAssumeNoMatchForDataFile), acceptPrefixMatchesForDataFile);
```

Comment:
I would prefer to do this by changing the mapping function, e.g.
```
if (!forceAssumeNoMatchForDataFile)
    return reader.ceiling(searchKey, (pos, assumeNoMatch, compareKey) -> retrieveEntryIfAcceptable(searchOp, compareKey, pos, assumeNoMatch));
else
    return reader.ceiling(searchKey, (pos, assumeNoMatch, compareKey) -> pos >= 0
        ? retrieveEntryIfAcceptable(searchOp, compareKey, pos, assumeNoMatch))
        : assumeNoMatch || acceptPrefixMatchesForDataFile ? new RowIndexEntry(~pos) : null;
```
and keeping both `ceiling` and `retrieveEntryIfAcceptable` unchanged.